src/scrapers/hapoalim.ts

```diff
@@ -147,21 +168,20 @@ async function fetchAccountData(page: Page, baseUrl: string, options: ScaperOpti
   const endDateStr = moment().format(DATE_FORMAT);
 
   const accounts: TransactionsAccount[] = [];
-  for (let accountIndex = 0; accountIndex < accountsInfo.length; accountIndex += 1) {
-    const accountNumber = `${accountsInfo[accountIndex].bankNumber}-${accountsInfo[accountIndex].branchNumber}-${accountsInfo[accountIndex].accountNumber}`;
-
-    const balanceAndCreditLimitUrl = `${apiSiteUrl}/current-account/composite/balanceAndCreditLimit?accountId=${accountNumber}&view=details&lang=he`;
-    const balanceAndCreditLimit = await fetchGetWithinPage(page, balanceAndCreditLimitUrl) as BalanceAndCreditLimit;
-    const balance: number = balanceAndCreditLimit?.currentBalance;
 
-    const txnsUrl = `${apiSiteUrl}/current-account/transactions?accountId=${accountNumber}&numItemsPerPage=150&retrievalEndDate=${endDateStr}&retrievalStartDate=${startDateStr}&sortCode=1`;
-    const txnsResult = await fetchPoalimXSRFWithinPage(page, txnsUrl, '/current-account/transactions');
-    let txns: Transaction[] = [];
+  for (const account of accountsInfo) {
```

Comment:
out of curiosity, how does it works for you? await doesn't work in for...of... unless I'm mistaken
![CleanShot 2021-12-29 at 21 16 35@2x](https://user-images.githubusercontent.com/4751797/147696210-a93ef930-ed93-463f-adce-e63a2ca812b8.jpg)


![CleanShot 2021-12-29 at 21 16 43@2x](https://user-images.githubusercontent.com/4751797/147696221-625684d5-2b9f-4507-ae22-e825442aab8d.jpg)

Reply:
@daniel-hauser see comment above

Reply:
@esakal You probably run the code in a node version that doesn't support top level awaits, you can put the for loop inside a async function and it will work.